deviceconfig/config.go

```diff
@@ -0,0 +1,34 @@
+package deviceconfig
+
+// RootFile represents a file that is stored on a raw disk device.
+type RootFile struct {
+	// Name of the file to be read from kernel package.
+	// It is also the name of the update device handler (`/update/<Name>`).
+	Name string
+	// Offset on root disk device where this file/blob should be stored.
+	Offset int64
+	// Maximum length of file to be accepted during updates.
+	// `[Offset, Offset+MaxLength)` must not overlap for any 2 RootFiles for a device.
+	MaxLength int64
+}
+
+type DeviceConfig struct {
+	RootDeviceFiles []RootFile
+}
+
+var (
+	// DeviceConfigs contains a mapping from device model (github.com/gokrazy/gokrazy.Model) to device-specific config
+	DeviceConfigs = map[string]DeviceConfig{
+		// Odroid HC1/HC2/XU4
+		// (https://github.com/torvalds/linux/blob/c9e6606c7fe92b50a02ce51dda82586ebdf99b48/arch/arm/boot/dts/exynos5422-odroidhc1.dts#L14)
+		"Hardkernel Odroid HC1": {
+			// https://wiki.odroid.com/odroid-xu4/software/partition_table#tab__odroid-xu341
+			RootDeviceFiles: []RootFile{
+				{"bl1.bin", 512, 15360},       // sectors 1 - 30
```

Comment:
Can you express all these numbers in multiples of a `const sectorSize = 512` please? E.g. `{"bl1.bin", 1*sectorSize, 30*sectorSize}` etc.